### ISSUE TITLE
App Removal due to SDK

### ISSUE THREAD

**Issue by Julian-Pfaff** [Pushwoosh/pushwoosh-android-sdk#181]
Hi there,

We received an Email today from Google App Store, which informed us about the removal of our app due to the PushWoosh SDK.

"Your app is using the Pushwoosh SDK/Library which collects Android ID and Location. Persistent device identifiers may not be linked to other personal and sensitive user data or resettable device identifiers."

I cannot find anything concerning this in the latest release notes nor in any open issues.
Is this something we can fix, or do we have to wait for a new version?

**Comment by lbonamini** [comment]
i have the same problem...which version do you use? 6.6.6?

**Comment by kwiky** [comment]
Hi 👋

Same problem, we need a fix 🙏

**Comment by wfhm** [comment]
@kwiky @lbonamini @Julian-Pfaff 

Thanks for reporting this. Pushwoosh SDK does not send Android ID to our backend, and it is only used under the hood for tags migration from really old SDK versions (back in 2017, Android ID was indeed used as a unique device identifier). This is the only place in the SDK code where Android ID is used, so we believe we can simply remove the migration part of the code.

This is really weird that Google started sending these messages today, as this part of the code had not changed for 4+ years, and was never reported as problematic before. 

We are already working on a fix, and the updated SDK version will be published shortly.

**Comment by wfhm** [comment]
@kwiky @Julian-Pfaff @lbonamini We've just rolled out the 6.6.7 version with Android ID usage completely removed, it should be available on maven shortly. Please let us know if the issue persists.

**Comment by Bejasc** [comment]
@wfhm when can I expect to see the package appearing in Visual Studio NuGet package manager? I can't see past 6.0.6 even with pre-releases ticked.

This is an absolute killer for my team - it's a bit of a worry that PushWoosh haven't been checking their compliance with the Google Play Developer Policies in 4+ years..

**Comment by wfhm** [comment]
@Bejasc Xamarin package update should be available by the end of this work week. 

> This is an absolute killer for my team - it's a bit of a worry that PushWoosh haven't been checking their compliance with the Google Play Developer Policies in 4+ years..

The migration logics mentioned above was actually implemented right after Google Play policy on Android ID usage was changed . Before that, we were using Android ID as HWIDs, and after that we moved to using randomly generated IDs, only using Android ID internally to migrate existing devices to the updated version. 
Google was seemingly fine with that change until now, so I believe they just applied more strict limitations on the existing rule without publicly announcing it. 

Also, our [PushOn demo application](https://play.google.com/store/apps/details?id=com.pushwoosh.pushon) was available in Play Store all this time, and we were never warned by Google about any policy violations.
